bcda/auth/backend.go

```diff
@@ -261,22 +226,48 @@ func getPublicKey() *rsa.PublicKey {
 	return rsaPub
 }
 
-func getJWTClaims(backend *JWTAuthenticationBackend, tokenString string) jwt.MapClaims {
+func (backend *JWTAuthenticationBackend) GetJWTClaims(tokenString string) jwt.MapClaims {
+	token, err := backend.GetJWTToken(tokenString)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if !token.Valid {
+		return nil
+	}
+
+	return token.Claims.(jwt.MapClaims)
+}
+
+func (backend *JWTAuthenticationBackend) GetJWTToken(tokenString string) (*jwt.Token, error) {
+
 	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
 		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
 			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
 		}
 
 		return backend.PublicKey, nil
 	})
+	return token, err
+}
 
+// Save a token to the DB for a user
+func (backend *JWTAuthenticationBackend) CreateToken(user bcdagorm.User) (bcdagorm.Token, error) {
+	db := database.GetGORMDbConnection()
+	tokenString, err := backend.GenerateTokenString(user.UUID.String(), user.Aco.UUID.String())
 	if err != nil {
 		panic(err)
 	}
-
-	if !token.Valid {
-		return nil
+	// Get the claims of the token to find the token ID that was created
+	claims := backend.GetJWTClaims(tokenString)
+	token := bcdagorm.Token{
+		UUID:   uuid.Parse(claims["id"].(string)),
+		UserID: user.UUID,
+		Value:  tokenString,
+		Active: true,
 	}
+	db.Create(&token)
```

Comment:
NB for the whole team: if we do start using this function as part of the `bcda create-token` command, we will be storing each generated token in our db.

Reply:
It doesn't yet.  It can very easily though.

Reply:
I left this function functionally unchanged so it will NOT persist a token just yet.   It probably should soon though.